src/components/Common/ConnectionBanner.tsx

```diff
@@ -1,10 +1,15 @@
 import React from "react";
 import { Button, Row, Col } from "react-bootstrap";
 
-export default function ConnectionBanner({ loading, connected, reconnect }) {
-  return loading || connected ? (
-    ""
-  ) : (
+type Props = {
+  loading: boolean;
+  connected: boolean;
+  reconnect: () => void;
+};
+
+function ConnectionBanner({ loading, connected, reconnect }: Props) {
+  if (loading || connected) return null;
```

Comment:
This should be one level up. 
The component should not be in charge of not rendering itself.